test/Constraints/protocols.swift

```diff
@@ -178,19 +178,19 @@ func existential(_ p: P) {
 }
 
 func staticExistential(_ p: P.Type, pp: P.Protocol) {
-  let _ = p() // expected-error{{value of type 'P.Type' is a protocol; it cannot be instantiated}}
-  let _ = p().bar // expected-error{{value of type 'P.Type' is a protocol; it cannot be instantiated}}
-  let _ = p().bar(1) // expected-error{{value of type 'P.Type' is a protocol; it cannot be instantiated}}
+  let _ = p() // expected-error{{initializing from a metatype value must reference 'init' explicitly}}
```

Comment:
Great! Thank you! It would be great to extract this logic for `onNonConstMetatype` into separate lambda and try to simplify all of the conditions, that would de-dup a lot of code.

Reply:
@xedin Done! :)